proto/viam/app/v1/app.proto

```diff
@@ -624,6 +627,21 @@ message DeleteRobotPartRequest {
   string part_id = 1;
 }
 
+message GetRobotAPIKeysRequest {
+  string robot_id = 1;
+}
+
+message RobotApiKey {
```

Comment:
[minor] capitalization doesn't match othoer stuff in this PR, this should be `RobotAPIKey`

Reply:
Might be worth calling this `APIKey` instead of `RobotAPIKey` so then i can add the optional `AuthorizationDetails` field in the response and reuse it. Thoughts?

Reply:
yeah either works for me -- makes sense to converge as they map to the same object

Reply:
hmm if authorizations are optional i think it makes it more difficult to know what to expect from each endpoint. i think `optional` should mean "will be returned if the value is available", not "will be returned depending on the context"

maybe this is a comment for APP-2589, but what do you think of having one message `APIKey` with the `key`, `id`, `created_on`, etc. and then returning the authorizations as a separate message? that way we can reuse `APIKey` here, because I agree it doesn't really make sense to have `RobotAPIKey` and `APIKey`. if we want two separate messages it should probably be something like `APIKey` and `APIKeyWithAuthorizations`

Reply:
sure yeah i think thats valid. I think for Roxy's PR it probably makes sense to leave as is and then we can chat and fix on mine. sound good?

Reply:
the only thing relevant to Roxy's PR is the name of the message. otherwise you might have to make a breaking change on yours. how do you guys feel about `RobotAPIKey` versus `APIKey`? i am leaning towards `APIKey` because there is nothing in this proto message that is particular to robot API keys imo

Reply:
oh yeah when i said "leave as is" i meant with `APIKey`. I think `APIKey` makes the most sense as well.